src/mngt/ocf_mngt_cache.c

```diff
@@ -547,33 +558,29 @@ static void _ocf_mngt_init_instance_load(
 static int _ocf_mngt_init_new_cache(struct ocf_cache_mngt_init_params *params)
 {
 	ocf_cache_t cache = env_vzalloc(sizeof(*cache));
+	int result;
 
 	if (!cache)
 		return -OCF_ERR_NO_MEM;
 
 	if (ocf_mngt_cache_lock_init(cache)) {
-		env_vfree(cache);
-		return -OCF_ERR_NO_MEM;
+		result = -OCF_ERR_NO_MEM;
```

Comment:
Now we will attempt to deinit uninitialized lock in this path. You need to add a separate label to handle this correctly.